### ISSUE TITLE
FastAPI + Djantic - OpenAPI spec

### ISSUE THREAD

**Issue by mmcardle** [jordaneremieff/djantic#38]
HI,

First of all, this is a great project, getting a lot of value from it, so thank you.

Having an issue generating an openapi specification when using it in combination with FastAPI, specifically when using Django model choices.

#### Django Model
```
from django.db import models

class SimpleModelWithChoices(models.Model):
    class ChoiceType(models.TextChoices):
        CHOICE1 = "CHOICE1"
        CHOICE2 = "CHOICE2"

    choice = models.CharField("Type", max_length=30, choices=ChoiceType.choices)
```

#### Djantic Spec

```
from djantic import ModelSchema

class Schema1(ModelSchema):
    class Config:
        model = SimpleModelWithChoices


class Schema2(ModelSchema):
    class Config:
        model = SimpleModelWithChoices
```

#### FastAPI
```
from fastapi import APIRouter, FastAPI

router = APIRouter()
application = FastAPI()

@application.get("path1", response_model=Schema1)
def get1(instance: Schema1):
    return instance

@application.get("path2", response_model=Schema2)
def get2(instance: Schema2):
    return instance

application.include_router(router)

assert application.openapi()
```

#### Error
```
.venv\lib\site-packages\fastapi\openapi\utils.py:387: in get_openapi
    definitions = get_model_definitions(

    def get_model_definitions(
        flat_models: Set[Union[Type[BaseModel], Type[Enum]]],
        model_name_map: Dict[Union[Type[BaseModel], Type[Enum]], str],
    ) -> Dict[str, Any]:
        definitions: Dict[str, Dict[str, Any]] = {}
        for model in flat_models:
            m_schema, m_definitions, m_nested_models = model_process_schema(
                model, model_name_map=model_name_map, ref_prefix=REF_PREFIX
            )
            definitions.update(m_definitions)
>           model_name = model_name_map[model]
E           KeyError: <enum 'ChoiceEnum'>

.venv\lib\site-packages\fastapi\utils.py:28: KeyError
```

I think it is due to the way FastAPI keeps a mapping of classes and uses them internally. I have a workaround in my project

```
def __fix_enums(new_type):

    for item in new_type.__dict__["__fields__"].values():
        if "enum.EnumMeta" in str(item.type_.__class__):
            enum_values = [(i.value, i.value) for i in item.type_]

            new_enum_type = Enum(f"{new_type.__name__}_{item.name}_Enum", enum_values, module=__name__)
            setattr(item, "type_", new_enum_type)

    return new_type
```

I would be happy to produce a PR to fix the issue if needed, Maybe adding some sort of uniqueness in the name of the enum class in  `djantic/fields.py` ?
```
            python_type = Enum(  # type: ignore
                f"{field.name.title().replace('_', '')}Enum",
                enum_choices,
                module=__name__,
            )
```

#### Library Versions
```
djantic = "^0.3.5"
Django = "^3.2.5"
fastapi = "^0.68.0"
```

**Comment by jordaneremieff** [comment]
Hi @mmcardle, glad to hear you're finding this project useful! I don't have a lot of time to look into this myself, but I could review/merge a PR to resolve this issue if you're willing to contribute this.

**Comment by jordaneremieff** [comment]
This is released in 0.4.0. Thanks!
